kibana/templates/deployment.yaml

```diff
@@ -175,3 +181,40 @@ spec:
 {{ toYaml .Values.extraContainers | indent 6 }}
       {{- end }}
       {{- end }}
+      initContainers:
+        - name: configure-kibana-token
+          image: "{{ .Values.image }}:{{ .Values.imageTag }}"
+          imagePullPolicy: "{{ .Values.imagePullPolicy }}"
+          # TODO add retries
+          command:
+            - sh
+            - -c
+            - curl --output /usr/share/kibana/config/tokens/{{ template "kibana.fullname" . }} --fail -XPOST --cacert /usr/share/kibana/config/certs/tls.crt -u "$(ELASTICSEARCH_USERNAME):$(ELASTICSEARCH_PASSWORD)" "{{ .Values.elasticsearchHosts }}/_security/service/elastic/kibana/credential/token/{{ template "kibana.fullname" . }}?pretty"
```

Comment:
There's two different tokens and I think we may be confusing the two:

enrollment token: 
- generated by bin/elasticsearch-create-enrollment-token
- when consumed via bin/kibana-setup configures kibana.yml:
```
elasticsearch.hosts: ["https://localhost:9200/"]
elasticsearch.serviceAccountToken: XXXXXXX
elasticsearch.ssl.certificateAuthorities:
  [/absolute/path/to/kibana-8.3.3/data/ca_1659104310110.crt]
xpack.fleet.outputs:
  [
    {
      id: fleet-default-output,
      name: default,
      is_default: true,
      is_default_monitoring: true,
      type: elasticsearch,
      hosts: ["https://localhost:9200/"],
      ca_trusted_fingerprint: XXXXXXX,
    },
  ]
```

service account token (used here):
- gets the value needed to set `elasticsearch.serviceAccountToken` in kibana.yml

Reply:
These 2 tokens are really confusing.

This makes things a lot more complicated because the token is generated from the kibana init container when the chart is already deployed. That means that we can't use it in the Helm template itself to modify the `kibana.yaml` config map or environment variable before the real kibana container is starting.

AFAIK, the only thing we can do is mount it as a file into the kibana container and run a command to use it while kibana process has already been started.

Reply:
I'm thinking that might work..can we send a sighup signal without interfering with the process monitor? https://github.com/elastic/kibana/issues/52756#issuecomment-827018758.  It sounds like that may not work for all configurations though, need to check.

Reply:
Another solution would be to override the Kibana container command in the Helm template to run a script that read the mounted file, parse the token and load it as an environment variable or edit the `kibana.yaml` file, then finally start kibana by running manually the [default image command](https://github.com/elastic/dockerfiles/blob/a405a4d692031b72cefcb8523bd464aa3221ec47/kibana/Dockerfile). 

I'd like to avoid if we can, because that any change in the default image command (in a different kibana version or for people using customized kibana image) would break the chart. However, I feel that that may be the only way to do it.

Reply:
Reloading the kibana config with `kill -HUP` doesn't seem to handle the `elasticsearch.serviceAccountToken` value in my tests. In addition, I discovered that the `kibana.yaml` config file is opened in RO, so editing it to add this parameter requires too much workarounds. I ended up using a custom entrypoint scripts as mentioned in my previous comment => 2319e09

I guess that we can still rework it if we change the kibana docker image entrypoint in a later version.